src/burningglass/01_data_profiling_complete.Rmd

```diff
@@ -57,7 +69,93 @@ data_profiling <- function(df){
 }
 ```
 
-```{r}
+```{r message=FALSE, warning=FALSE}
 data_profiling(pers)
 ```
 
+# Identify DC Metropolitan Area
+
+## MSA
+```{r message=FALSE, warning=FALSE, paged.print=FALSE}
+#library(zipcode)
+
+DC_pers <- pers%>%
+  filter(grepl("47900", msa) == T)
+
+matches <- regmatches(DC_pers$msa, gregexpr("[[:digit:]]+", DC_pers$msa))
+msa_list <- unique(unlist(matches))
+msa_DC <- c("47900", "548")
+msa_not_DC <- setdiff(msa_list, msa_DC)
+
+dup_detect <- c()
+for (i in (1:length(matches))){
+  item <- matches[[i]]
+  if(sum(item %in% msa_not_DC == T) > 2){
+    dup_detect[i] = T
+  }else{
+    dup_detect[i] = F
+  }
+}
+
+```
+
+We identified 687129 people who haved listed "47900" in their msa, 0.30 % (N=2093) of people had duplicated entries in msa. Among people who only reported "47900" in their msa, there are 5033 unique self-reported zipcodes. Then, I matched the zipcode with zipcodes on US Zip Codes Database (using authoritative sources including the U.S. Postal Service™, U.S. Census Bureau, National Weather Service, American Community Survey, and the IRS). 28.00% of these 5033 zipcodes (N=1409) don't match with the zipcode database. 
```

Comment:
For MSA (in case you want to get a sense of the co-listed MSA...
```
library(maditr)
msa <- fread(input = "https://www2.census.gov/programs-surveys/cbp/technical-documentation/reference/metro-area-geography-reference/msa_county_reference17.txt")
# https://www2.census.gov/programs-surveys/cbp/technical-documentation/reference/state-county-geography-reference/georef17.txt for States + Counties (FIPS)
```

I have worked a lot with matching ZIP code data... Does the BLS do ZIP codes or census like (FIPS? or ZCTAs)?

Reply:
I'm not sure about BLS zip code. I got the dataset from https://simplemaps.com/data/us-zips. It built from Census Bureau data.

Reply:
I would comment the source (maybe put the link there as a reference?)

I guess I just don't understand what we get from using the ZIP code data... MSA do go across states so absent a state variable the ZIP code could help impute the state for example for example... I believe that's what you are doing from the comments below.

The file above gives you a mapping by MSA and all county / states included in the MSA (if only one state you can make the call directly). For MSA 47900 the ZIP code would be a good way to input the state.